### PR TITLE
Symlinks

### PR DESCRIPTION
**Pull request by constellates**
Fix for #92. I added better error handling for orphaned symlinks and otherwise unreadable files. Node has a variety of tools for evaluating files and symlinks, but I have not found the same for the browser. So node will actually check if a file is a symlink and inaccessible. The browser just checks if a file is readable and returns a more generic issue.

Also fixed a couple missing variable declarations that were causing errors in the browser.

### CODE REVIEW

**Comment by chrisgorgo** [comment]
LGTM - @hanke could you test it?

**Comment by constellates** [comment]
I merged in the most recent master. I'll comment again when I'm done re-testing the fix.

**Comment by constellates** [comment]
This has been tested and is still working. I also fixed a couple missing variable declarations that got added in when I merged the recent master. Node is lenient enough to assume we want them defined but they break the browser.

**Comment by constellates** [comment]
Up to date with current master.

File: utils/files.js [hunk 1]
```diff
@@ -279,4 +294,28 @@ function relativePath (file) {
...
 
+/**
+ * Test File
+ *
+ * Takes a file and callback and tests if it's viable for
+ * reading. Callsback with an error and stats if it isn't
+ * or null and stats if it is.
+ */
+function testFile (file, callback) {
+    fs.lstat(file.path, function (statErr, stats) {
```
Comment (chrisgorgo) [inline utils/files.js:305]: do those calls work in the browser?

File: utils/files.js [hunk 2]
```diff
@@ -279,4 +294,28 @@ function relativePath (file) {
...
+        fs.access(file.path, function (accessErr) {
+            if (!accessErr) {
+                callback(null, stats);
+                return;
+            } else {
+                if (stats.isSymbolicLink()) {
+                    callback(new Issue({code: 43, file: file}), stats);
```
Comment (chrisgorgo) [inline utils/files.js:312]: This discards all symbolic links - those that point to real files as well as dangling ones.

**Comment by chrisgorgo** [comment]
I jumped the gun and merged it just to discover it has a critical bug: it raises and error for any symbolic links not only dangling ones. I had to revert the merge here: https://github.com/INCF/bids-validator/commit/58156e52d2a1d1c20caae9294a55fd7070b1cfa7. Sorry for the mess.

File: utils/files.js (hunk 1 above)
Comment (constellates) [inline utils/files.js:305]: Those calls don't work in the browser, but that function is only ever called in node. We first check if `fs` exists. I'll update this to fix the issue you found below and incorporate master and resubmit.

**Pull request merged**